tools/scripts/rebuildAnimationLibraryManifest.rb

```diff
@@ -0,0 +1,217 @@
+#!/usr/bin/env ruby
+#
+# Regenerates an animation library manifest by reading the contents of S3 bucket
+# cdo-animation-library.
+#
+# For usage, run ./rebuildAnimationLibraryManifest.rb --help
+#
+# The manifest has three important jobs:
+# 1. Provide the metadata used to render spritesheets as animations.
+# 2. Point to particular versions of the spritesheets on S3.
+# 3. Provide a precomputed search tree for quick animation searches by keyword.
+#
+# The expected output of this file is a new animationLibrary.json which is
+# consumed by the apps build.  You'll need to manually check this file into git.
+#
+# See also: The Animation Library Tech Spec (requires login):
+# https://docs.google.com/document/d/18-LVuvKd0jKTUiGo5GYReUWM5oFWCyKRyEQURJ5HCOM/edit
+#
+# TODO: Optimize: Read existing manifest and don't do full object reads for
+#       entries whose modify date hasn't changed.
+require 'aws-sdk'
+require 'optparse'
+require_relative '../../deployment'
+
+DEFAULT_S3_BUCKET = 'cdo-animation-library'
+DEFAULT_OUTPUT_FILE = 'apps/src/gamelab/animationLibrary.json'
+
+class ManifestBuilder
+  def initialize(options)
+    @options = options
+  end
+
+  #
+  # Main Entry Point
+  # See end of file for possible options, parsed from command line
+  #
+  def rebuild_animation_library_manifest
+    animations_by_name = {}
+
+    # Connect to S3 and get a listing of all objects in the animation library bucket
+    info 'Reading object list from S3...'
+    bucket = Aws::S3::Bucket.new(DEFAULT_S3_BUCKET)
+    bucket.objects.each do |object_summary|
+      animation_name = object_summary.key[/^[^.]+/]
+      extension = object_summary.key[/(?<=\.)\w+$/]
+      verbose <<-EOS.unindent
+        #{bold object_summary.key}
+          #{object_summary.last_modified} | #{object_summary.size}
+      EOS
+      # Push into animations collection
+      animations_by_name[animation_name] ||= {}
+      animations_by_name[animation_name][extension] = object_summary
+    end
+
+    # TODO: Validate that keys are unique
```

Comment:
There's a good number of validation TODOs in here that I'd like to tackle in a separate PR as prep for setting @poorvasingal loose with this script and the Kenney.nl assets.